eeveebeats/EeveeBeats
# implemented threejs

## Pull Request - React-Redux Setup and Three.js Test Page

### Changes Introduced:
1. **React-Redux Boilerplate Setup:**
   - Implemented the foundational structure for a React-Redux application.
   - Added necessary dependencies including `react-redux`, `redux-thunk`, and `redux-devtools-extension` for enhanced development experience.
   - Configured the Redux store and integrated it with the React application.
   - Created a basic example of Redux actions, reducers, and a store to demonstrate data flow.

2. **Three.js Test Page Implementation:**
   - Integrated Three.js with our React application to establish a foundation for 3D rendering.
   - Created a new `ThreeTestPage` component as a playground for our developers to experiment with Three.js features.
   - The test page includes a basic 3D scene with fundamental elements like a camera, renderer, and sample geometry.
   - Added comments and documentation on how to use the Three.js setup for educational purposes and future development.

### Purpose:
The addition of React-Redux boilerplate is aimed at streamlining state management across the application, while the Three.js test page serves as a starting point for integrating sophisticated 3D graphics into our project. This setup allows other developers on the team to jumpstart development and experimentation with 3D elements.

### How to Test:
- For Redux:
  - Navigate to the Redux example component.
  - Observe the Redux state management in action through actions and reducers.

- For Three.js:
  - Access the `ThreeTestPage` component.
  - Experiment with modifying the Three.js scene by adding or changing elements.

### Notes:
- Further enhancements and more complex Three.js implementations will follow as separate tasks/PRs.
- Any feedback or suggestions on the current setup are greatly appreciated.

@team, kindly review the changes and provide your insights.
